### ISSUE TITLE
Work with package maintainers to get 3.0 into major platform distributions

### ISSUE THREAD

**Issue by nicolan** [cyrusimap/cyrus-imapd#1819]
We provide install instructions from packages for a bunch of major platforms (RHEL, Debian, Ubuntu, and others).

A quick glance shows that most of them mention old Cyrus (2.4 or earlier) and old OS distros.

Needs someone with access to these platforms to refresh the docs.

- [ ] Debian - @nicolan
- [ ] Ubuntu
- [ ] Fedora - @jasontibbitts 
- [ ] RHEL - @hagedose / @jasontibbitts
- [ ] CentOS - @hagedose 
- [ ] OpenSUSE

**Comment by hagedose** [comment]
I have access to RHEL and CentOS systems and would be willing to help. What can I do?

**Comment by nicolan** [comment]
@hagedose Brilliant! The current documentation files for the package distro installs are in docsrc/imap/download/installation/distributions/

If you can do an install from packages on each OS and update the documentation to reflect any changes in libraries or commands, that would be great. I'm also not sure what version of Cyrus each platform is shipping, so insight there would be appreciated too!

Remove the section on Other Versions of Cyrus IMAP as it refers to using the Open Build Systems packages which don't exist any more.

The restructured text formatting is pretty self explanatory, but if you want to check the formatting and build the html, there's instructions http://cyrusimap.org/dev/imap/developer/documentation.html#contribute-docs

**Comment by hagedose** [comment]
I'm having a problem setting up the development environment for documentation. I have installed sphinx and Pod::POM::View::Restructured on a CentOS 7 system, and then run autoreconf and configure (configure finds the packages). According to the page you cite I should run "make clean init man html" next. Make tells me that there is no target named "init". When I leave out init, the other targets appear to work, but it doesn't look like anything is done with the files in docsrc/imap/download/installation/distributions/. So I tried to convert the restructuredText files myself using rst2man and rst2html, but that doesn't seem to work either:

```
$ rst2html centos.rst centos.html
centos.rst:22: (ERROR/3) Unknown interpreted text role "command".
centos.rst:28: (ERROR/3) Unknown interpreted text role "command".
centos.rst:36: (ERROR/3) Unknown interpreted text role "manpage".
centos.rst:39: (ERROR/3) Unknown interpreted text role "command".
centos.rst:39: (ERROR/3) Unknown interpreted text role "command".
centos.rst:45: (ERROR/3) Unknown interpreted text role "manpage".
centos.rst:48: (ERROR/3) Unknown interpreted text role "command".
centos.rst:54: (ERROR/3) Unknown directive type "todo".

.. todo::
    If this does not succeed, see ref:`sasl-troubleshooting-saslauthd`.

centos.rst:60: (ERROR/3) Unknown interpreted text role "command".
centos.rst:60: (ERROR/3) Unknown interpreted text role "command".
centos.rst:68: (ERROR/3) Unknown interpreted text role "command".
centos.rst:91: (ERROR/3) Unknown interpreted text role "ref".
centos.rst:9: (ERROR/3) Undefined substitution referenced: "imap_tikanga_stock_version".
centos.rst:10: (ERROR/3) Undefined substitution referenced: "imap_santiago_stock_version".
centos.rst:11: (ERROR/3) Undefined substitution referenced: "imap_maipo_stock_version".
```

Wha am I doing wrong?

**Comment by elliefm** [comment]
There's two Makefiles for the docs, and this is causing the confusion.

In the top level project directory, there is the Makefile that is generated by autoconf.  This makefile accepts targets `doc-html`, `doc-text`, and `man` to build the html, plain text, and man documentation respectively.  It also accepts a target `doc` which builds all three (i.e. `make doc`).  This is the preferred way of building the documentation, as it is properly integrated with the project.  The generated documentation lands in `doc/html/`, `doc/text/` and `man/` respectively.

In the `docsrc/` directory there is a static legacy Makefile, which accepts targets `init`, `man`, `html`, and produces its output in `docsrc/build/man/`, `docsrc/build/html/`, etc.  This is leftover from before the doc building was integrated into autoconf.  It's still used for generating the cyrusimap.org website for reasons, and I believe it might also work on Windows?  But generally it shouldn't be used these days, and will hopefully go away sometime.

Sounds like the "how to build documentation" documentation could use an update too!

**Comment by hagedose** [comment]
Thanks, I wasn't aware of that second Makefile. However, while it generates HTML from the .rst files, there are many error messages in the result. I haven't yet made any changes to the source file myself. This is one of the errors:

> System Message: ERROR/3 (centos.rst, line 22); backlink
> Unknown interpreted text role "command".

There are others for "manpage" and "ref". There are also these messages:

> Undefined substitution referenced: "imap_tikanga_stock_version"

Looks like I'm still missing something.

**Comment by elliefm** [comment]
cool :)

**Comment by hagedose** [comment]
Now that my setup is working I wonder what the actual goal for these documents is. As they stand, they only document how to install the packages that are supplied by the distributions. The only necessary change for CentOS and RHEL would then be to remove the entries for release 5, which reaches its EOL at the end of March. But that would mean the most modern version of Cyrus is still 2.4.

Do we really even want to document that anymore?

**Comment by elliefm** [comment]
I think the idea is how to install _3.0_ from packages on these platforms -- which is meaningless until these platforms package up 3.0.

But conversely, there's a chicken-egg scenario here in that platforms will probably package the newer releases faster if there's already documentation about which dependencies/versions they require and such.  But that documentation lives elsewhere.

So, I guess I don't really know what the goal is here.  Maybe this section should only include platforms that package 3.0 -- in which case, I believe it would currently be empty?

But also, presumably, these platforms already document how to install stuff from their package management, soooo... I dunno.

This feels like a conversation to have with various distro's package maintainers, I suspect?

**Comment by hagedose** [comment]
I seriously doubt there will be official packages anytime soon. I would suggest there should be packages maintained by the Cyrus community. For the past several years I have used packages maintained by Simon Matter, because those provided by Red Hat were always hopelessly outdated. However, Simon's RPMs are stuck at 2.4.18. I'm not sure if he would be interested to continue the work. I would definitely be willing to help, but I'm afraid being the sole maintainer for CentOS and RHEL would overwhelm me.

**Comment by nicolan** [comment]
I was originally aiming to document whatever the latest maintained version for each OS was. With noted caveats in the documentation if the package was older than latest stable (2.5.10 for now).

But if most of the platform packages are significantly out of date, perhaps we just remove those sections altogether for now, until we know they've been updated and a maintainer is known.

**Comment by jasontibbitts** [comment]
I just noticed this.

I actually would like to try and get 3.0 into EPEL7.  This would require a bit of work as it would not be permitted for that package to conflict (at the file level) with the one provided by Red Hat.  That means that executables, service names, configuration directories, etc. need to be renamed.  I would welcome any assistance with that kind of thing.  Or, of course, if the cyrus developers feel that such a package would be too confusing, I could punt and do something else.

But first I have to get 3.0 into Fedora....

**Comment by dilyanpalauzov** [comment]
Can this be closed, as 3.0 is outdated now?
